lib/schemas.js

```diff
@@ -73,7 +73,24 @@ function normalizeSchema (routeSchemas, serverOptions) {
   for (const key of SCHEMAS_SOURCE) {
     const schema = routeSchemas[key]
     if (schema && !isCustomSchemaPrototype(schema)) {
-      routeSchemas[key] = getSchemaAnyway(schema, serverOptions.jsonShorthand)
+      if (key === 'body') {
+        const contentProperty = routeSchemas.body.content
+        if (contentProperty) {
+          const keys = Object.keys(contentProperty)
+          for (let i = 0; i < keys.length; i++) {
+            const mediaName = keys[i]
+            const contentSchema = contentProperty[mediaName].schema
+            if (!contentSchema) {
+              throw new FST_ERR_SCH_CONTENT_MISSING_SCHEMA(mediaName)
+            }
+            routeSchemas.body.content[mediaName].schema = getSchemaAnyway(contentSchema, serverOptions.jsonShorthand)
+          }
+        } else {
+          routeSchemas.body = getSchemaAnyway(routeSchemas.body, serverOptions.jsonShorthand)
+        }
+      } else {
+        routeSchemas[key] = getSchemaAnyway(schema, serverOptions.jsonShorthand)
+      }
```

Comment:
Too many branches IMHO

```suggestion
      if (key === 'body') {
        const contentProperty = schema.content
        if (contentProperty) {
          const keys = Object.keys(contentProperty)
          for (let i = 0; i < keys.length; i++) {
            const mediaName = keys[i]
            const contentSchema = contentProperty[mediaName].schema
            if (!contentSchema) {
              throw new FST_ERR_SCH_CONTENT_MISSING_SCHEMA(mediaName)
            }
            routeSchemas.body.content[mediaName].schema = getSchemaAnyway(contentSchema, serverOptions.jsonShorthand)
          }
          continue
        }
      }
      routeSchemas[key] = getSchemaAnyway(schema, serverOptions.jsonShorthand)
```